manager/controllers/app/moduleinstance.go

```diff
@@ -461,14 +461,21 @@ func moduleAPIToService(api *app.ModuleAPI, scope app.CapabilityScope, appContex
 	releaseName := utils.GetReleaseName(appContext.Name, appContext.Namespace, instanceName)
 	blueprintNamespace := utils.GetBlueprintNamespace()
 
-	values := map[string]interface{}{
-		"Release": map[string]interface{}{
+	values := make(map[string]interface{})
+	if len(appContext.Labels) == 0 {
+		values["Release"] =
+			map[string]interface{}{
+				"Name":      releaseName,
+				"Namespace": blueprintNamespace,
+			}
+	} else {
+		values["Release"] = map[string]interface{}{
 			"Name":      releaseName,
 			"Namespace": blueprintNamespace,
-		},
-		"Values": map[string]interface{}{
+		}
+		values["Values"] = map[string]interface{}{
 			"labels": appContext.Labels,
-		},
+		}
 	}
```

Comment:
Looking forward we may add other attributes to Values so we may want to improve this to be a bit cleaner.
Consider either initializing as an empty map when the labels is nil, or using structs with `omitempty` in the json tag.

Reply:
updated code